modules/src/clients/ics07_tendermint/client_state.rs

```diff
@@ -127,11 +119,16 @@ impl ClientState {
         }
     }
 
-    pub fn with_set_frozen(self, h: Height) -> Self {
-        Self {
-            frozen_height: h,
-            ..self
+    pub fn with_set_frozen(self, h: Height) -> Result<Self, Error> {
```

Comment:
Since this PR already includes a breaking change, let's rename this to

```suggestion
    pub fn with_frozen_height(self, h: Height) -> Result<Self, Error> {
```